flink-tests/src/test/java/org/apache/flink/test/checkpointing/ChangelogPeriodicMaterializationTestBase.java

```diff
@@ -234,6 +235,25 @@ protected JobID generateJobID() {
         return Collections.emptySet();
     }
 
+    private static Optional<CheckpointMetadata> getMostRecentCompletedCheckpointMetadata(
+            File checkpointFolder) throws IOException {
+        try {
```

Comment:
Thanks for the fix @masteryhx. I think it's correct.
However, it only solves the problem for the current test. There are a couple of more tests using the same code.
Maybe it makes sense to fix it in `TestUtils`?

Reply:
Sure, I check the exception in `hasMetadata` of `TestUtils`, it should solving the problem of other test.